docs/source/en/tutorials/typescript.md

```diff
@@ -631,3 +648,107 @@ export default class NewsService extends Service {
+
+If the egg's plugin has the right declaration, we need to import it exclipitly and ts can load the related object.
+
+If you use 'egg-ts-helper', it will automatically generate the exclipit importing declarations according to what plugins you've enabled in the application. If you don't use that, you have to import the declaration of plugins manually.
+
+```typescript
+// typings/index.d.ts
+
+import 'egg-dashboard';
+```
+
+**Notice: You MUST use 'import' in 'd.ts', because most of egg's plugins are without main entry points. There'll be errors occuring if you import directly in ts.**
+
+### 'paths' is invalid in 'tsconfig.json'
+
+Strictly speaking, this has nothing to do with 'egg' but with many people's questions, and we'll give our answer to it. The reason is tsc WON'T convert the import path when compiling ts to js, so when you config paths in 'tsconfig.json' and if you use 'paths' to import the related modules, you are running the high risk that you cannot find them when compiled to js.
+
+The solution is either you don't use 'paths', or you can ONLY import some declarations instead of detailed values. Another way is that you can use [tsconfig-paths](https://github.com/dividab/tsconfig-paths) to hook the process logic in node's path module to support 'paths' in 'tsconfig.json'.
+
+You can directly import 'tsconfig-paths' in 'config/plugin.ts', because 'plugin.ts' is ALWAYS loaded firstly in both App and Agent.
+
+```typescript
+// config/plugin.ts
+
+import 'tsconfig-paths/register';
+
+...
+```
+
+### How to write unit tests for declarations of egg's plugins?
+
+Many contributors don't know how to write unit tests for their plugin's declaration files, so we also have a discussion about it here:
+
+When you finish writing a declaration for an egg's plugin, you can write your own application in `test/fixures` (you can refer https://github.com/eggjs/egg-view/tree/master/test/fixtures/apps/ts ). Do remember to add 'paths' configs into 'tsconfig.json' to do imports in the fixture. Take 'egg-view' as an example:
+
+```json
+    "paths": {
+      "egg-view": ["../../../../"]
+    }
+```
+
+Do remember DO NOT CONFIG `"skipLibCheck": true` in the 'tsconfig.json', and if you set it to true, tsc will ignore the type checks when compiling, and there's no meaning for unit tests for your plugin's declarations.
+
+In the end, add a test case to check whether your declaration works properly or not. See 'egg-view' example below:
+
+```js
+  describe('typescript', () => {
+    it('should compile ts without error', () => {
+      return coffee.fork(
+        require.resolve('typescript/bin/tsc'),
+        [ '-p', path.resolve(__dirname, './fixtures/apps/ts/tsconfig.json') ]
+      )
+        // .debug()
+        .expect('code', 0)
+        .end();
+    });
+  });
+```
+
+Some other unit test projects as your references:
+
+ - https://github.com/eggjs/egg
+ - https://github.com/eggjs/egg-view
+ - https://github.com/eggjs/egg-logger
```

Comment:
Please wrap these links with `[]` like `[https://github.com/eggjs/egg](https://github.com/eggjs/egg)`, and change the `zh-cn` as the same if you don't mind